ykcompile/src/lib.rs

```diff
@@ -250,41 +258,99 @@ impl TraceCompiler {
             ));
         };
 
-        // FIXME implement support for arguments.
-        if !args.is_empty() {
-            return Err(CompileError::Unimplemented("call with args".to_owned()));
+        if args.len() > 6 {
+            return Err(CompileError::Unimplemented(
+                "call with spilled args".to_owned(),
+            ));
         }
 
-        // Generally speaking, we don't know what the callee will clobber, so we save our registers
-        // before doing the call. For now we assume that the caller is OK with RAX being clobbered
-        // though.
-        for reg in 1..X86_86_NUM_REGS {
+        // Figure out where the return value (if there is one) is going.
+        let dest_reg: Option<u8> = if let Some(d) = dest {
+            Some(self.local_to_reg(d.local)?)
+        } else {
+            None
```

Comment:
We currently assume that there is always a return variable which is guarded by an assert (https://github.com/softdevteam/yk/blob/master/yktrace/src/tir.rs#L177). So technically there's no need for this else branch here.

Reply:
Ah wait. That assert is only for inlined function calls. Though we could make the same assumption for native calls, since I haven't yet managed to produce code that generates calls that don't have a destination. Given `dest` is an `Option` there must be a way, but until we know for sure, we can just assume there isn't, and hopefully this will blow up some day on some example.

Reply:
So `else { panic() }`?

Reply:
This will all be rewritten once we implement spilling, so it's not super important. But you can just add an `assert!(newdest.is_some());` to `tir.rs` before converting the `Call`, and then we can get rid of the else branch entirely (or put `unreachable` in it).

Reply:
Actually now that I look at the `Call` conversion: how come we don't rename its return destination? https://github.com/softdevteam/yk/blob/master/yktrace/src/tir.rs#L196

Reply:
We do, I think. I'm not sure that's relevant.

`dest` is an option, so we have do something with the possibility of it being `None`, even if we know that's impossible. I'm happy with `unreachable!`. Or we could get rid of `Option` (later)?

Reply:
Removing the `Option` might be a bit premature as I don't yet know if there are cases where it can't be `None`. We can do `unreachable!` or you just unwrap it (which will also crash if the "impossible" happens). That's up to you.

Reply:
> We do, I think. I'm not sure that's relevant.

I don't see it. Notice how we rename the destination for the `Enter` case (https://github.com/softdevteam/yk/blob/master/yktrace/src/tir.rs#L170) but not for the `Call` case.

Reply:
It needs to happen for both.

Do we just need to lift that rename out of the `if let...`?

Reply:
If you want to finish this PR first, we can do this in a separate PR. We just need to copy this